src/include/stir/listmode/LmToProjDataAbstract.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+//
+//
+/*!
+  \file 
+  \ingroup listmode
+
+  \brief Abstract base class for listmode to projection data conversion.
+ 
+  \author Richard Brown
+  
+*/
+/*
+    Copyright (C) 2020, University College of London
+    This file is part of STIR.
+
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+#include "stir/ParsingObject.h"
+
+START_NAMESPACE_STIR
+
+/*!
+  \ingroup listmode
+  \ingroup NiftyPET
+
+  \brief This class is the abstract base class fir binning listmode data to projection data,
+  i.e. (3d) sinograms.
+
+  It provides the basic machinery to go through a list mode data file,
+  and write projection data for each time frame. 
+*/
+
+class LmToProjDataAbstract : public ParsingObject
+{
+public:
+
+    /// Destructor
+    virtual ~LmToProjDataAbstract() {}
+
+    /// Set up
+    virtual bool set_up() { return false; }
```

Comment:
`set_up` should normally return `Succeeded` I guess

Reply:
I did this to match SIRF which inherits from this class. But I should switch this one to `Succeeded` and do the same on the SIRF side.

Reply:
I haven't followed carefully with SIRF `set_up` conventions are. I'd have hoped that it is a `void`, and throws (that's according to the SIRF guidelines I believe). @evgueni-ovtchinnikov could confirm.

In any case, this is STIR, and we have to do what other STIR `set_up` functions do :-)

I believe that in STIR we should also gradually switch from "checking return values" to throwing (as people generally don't check the return anyway, certainly not in Python/MATLAB), but that's for the future.